asciidoc-confluence-publisher-client/src/main/java/org/sahli/asciidoc/confluence/publisher/client/http/ConfluenceRestClient.java

```diff
@@ -370,15 +377,43 @@ private static void closeInputStream(InputStream inputStream) {
         }
     }
 
-    private static CloseableHttpClient defaultHttpClient() {
+    private static CloseableHttpClient defaultHttpClient(boolean disableSSLVerfication) {
         RequestConfig requestConfig = RequestConfig.custom()
                 .setConnectionRequestTimeout(20 * 1000)
                 .setConnectTimeout(20 * 1000)
                 .build();
 
-        return HttpClients.custom()
-                .setDefaultRequestConfig(requestConfig)
-                .build();
+        HttpClientBuilder builder = HttpClients.custom()
+                .setDefaultRequestConfig(requestConfig);
+
+        if (disableSSLVerfication) {
+            builder.setSSLContext(emptySSLContext());
+        }
+        return builder.build();
+    }
+
+    private static SSLContext emptySSLContext() {
```

Comment:
Using the `SSLContextBuilder` from Commons, you can create a trust-all SSL context in an easier way:

```
    private static SSLContext trustAllSslContext() {
        try {
            return new SSLContextBuilder()
                    .loadTrustMaterial((certificate, authType) -> true)
                    .build();
        } catch (Exception e) {
            throw new RuntimeException("Could not create trust-all SSL context", e);
        }
    }
```

Also, please install a no-op hostname verifier:

```
        if (disableSSLVerfication) {
            builder.setSSLContext(trustAllSslContext());
            builder.setSSLHostnameVerifier(new NoopHostnameVerifier());
        }
```